File: frontend/src/app/main/dto/games/stats/GameStats.ts
import {PlayerDetails} from "../../all/PlayerDetails";
import {CorrelationResult} from "./CorrelationResult";
import {GameStatsMetrics} from "./GameStatsMetrics";
import {ChartDataHistory} from "./ChartDataHistory";
import {GameStatsStreakResult} from "./GameStatsStreakResult";

export interface GameStats {
    player: PlayerDetails;
    gamesPlayed: number;
    winRate: number;
    winStreaks: GameStatsStreakResult;
    scoreMetrics: GameStatsMetrics<number>,
    durationMetrics: GameStatsMetrics<string>,
    nemesis: PlayerDetails | null;
    victim: PlayerDetails | null;
    rival: PlayerDetails | null;
    companion: PlayerDetails | null;
    correlations: CorrelationResult[];
    chartData: ChartDataHistory[];
    gameSpecific: any;
}

File: backend/src/main/java/net/zorphy/backend/main/game/dto/stats/GameStats.java
package net.zorphy.backend.main.game.dto.stats;

import net.zorphy.backend.main.game.dto.stats.chart.ChartData;
import net.zorphy.backend.main.game.dto.stats.correlation.CorrelationResult;
import net.zorphy.backend.main.game.dto.stats.metrics.GameStatsMetrics;
import net.zorphy.backend.main.game.dto.stats.streaks.GameStatsStreakResult;
import net.zorphy.backend.main.player.dto.PlayerDetails;

import java.time.Duration;
import java.util.List;


public record GameStats(
        PlayerDetails player,
        int gamesPlayed,
        double winRate,
        GameStatsStreakResult winStreaks,
        GameStatsMetrics<Double> scoreMetrics,
        GameStatsMetrics<Duration> durationMetrics,
        PlayerDetails nemesis,
        PlayerDetails victim,
        PlayerDetails rival,
        PlayerDetails companion,
        List<CorrelationResult> correlations,
        List<ChartData> chartData,
        GameSpecificStats gameSpecific
) {
}
